resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/domain/CompletenessEntity.java

```diff
@@ -88,9 +100,15 @@ public String getEntityId() {
         return entityId;
     }
 
-    public Integer getId() {
-        this.id = hashCode();
-        return id;
+    /**
+     * Creates a hash value for the passed parameters
+     *
+     * @param entityType the Entity type enum object
+     * @param entityId The Id of the entity, which eventually becomes a hash of the type and Id as a primary key
+     * @return the hash value used for the id, made up of the entity type and entity id values
+     */
```

Comment:
I think this should be left as is for now. Using this means that each entry is unique and based on the value that's stored.

Reply:
I think this has the potential to be quite nasty later on so. Since it wasn't done as part of this ticket I've raised https://jira.tdx.oneit.gov.uk/browse/PAL-964 to look at later.